### PR TITLE
feat: add support for dual stack clusters

### PR DESCRIPTION
**Pull request by M0NsTeRRR**
Support dual stack clusters :)

### CODE REVIEW

**Comment by rdettai** [comment]
Thanks for this contribution! Can you tell us a bit more about a use case when this is useful? Even though it's just an extra feature, I'm a bit concerned by the ever growing number of tuning nobs on the chart!

File: charts/quickwit/values.yaml [hunk 1]
```diff
@@ -426,6 +426,11 @@ prometheusRule:
 service:
   type: ClusterIP
 
+  # -- Set the ip family policy to configure dual-stack see [Configure dual-stack](https://kubernetes.io/docs/concepts/services-networking/dual-stack/#services)
+  ipFamilyPolicy: ""
+  # -- Sets the families that should be supported and the order in which they should be applied to ClusterIP as well. Can be IPv4 and/or IPv6.
+  ipFamilies: []
+
```
Comment (rdettai) [inline charts/quickwit/values.yaml:433]: Not sure what the best practice is here.

```suggestion
  # -- Set the ip family policy to configure dual-stack. See [Configure dual-stack](https://kubernetes.io/docs/concepts/services-networking/dual-stack/#services)
  # ipFamilyPolicy: "PreferDualStack"
  # -- Sets the families that should be supported and the order in which they should be applied to ClusterIP as well. Can be IPv4 and/or IPv6.
  # ipFamilies: ["IPv4","IPv6"]

```

**Comment by M0NsTeRRR** [comment]
Yes, indeed. Some users, like myself, operate dual-stack clusters with both IPv4 and IPv6 deployed. Quickwit for now only supports IPv4 services in dual stack configuration, as IPv4 takes precedence over IPv6 and this cannot be altered. The introduction of the `ipFamilies` feature allows users to prioritize IPv6 over IPv4. `ipFamilyPolicy` will allow user to choose if what they want to assign. Why is this significant? Well, IPv6 and IPv4 represent distinct networks, each with its own latency and speed characteristics, which can vary depending on the internet service providers. Moreover, enabling this feature facilitates the seamless operation of existing IPv6 and IPv4 network applications together.

Additionally, it's worth noting that Quickwit pods already possess IPv4, and only IPv6 services are affected, as detailed in the Kubernetes documentation here: [link](https://kubernetes.io/docs/concepts/services-networking/dual-stack/).

Importantly, the current implementation does not constitute a breaking change. Rather, it allow concerned users to opt into IPv6 support if they desire.

File: charts/quickwit/values.yaml (hunk 1 above)
Comment (M0NsTeRRR) [reply charts/quickwit/values.yaml:433]: We could comment it out to indicate that it's not a required configuration if you prefer. However, I believe it's better to retain the current value to demonstrate the default behavior.

**Comment by guilload** [comment]
Ok to ship this feature as long as it remains invisible to users who don't care about IPv6.

File: charts/quickwit/values.yaml (hunk 1 above)
Comment (rdettai) [reply charts/quickwit/values.yaml:433]: the main benefit of commenting out I think is that when we test it, we test that it is compatible with old configuration that didn't have the value! 

> I believe it's better to retain the current value to demonstrate the default behavior

What is the default behavior actually? `""` and `[]` are pretty confusing to me.

Comment (M0NsTeRRR) [reply charts/quickwit/values.yaml:433]: > the main benefit of commenting out I think is that when we test it, we test that it is compatible with old configuration that didn't have the value!
As you want :)

> What is the default behavior actually? "" and [] are pretty confusing to me.
They are not set as they are treated as empty value for gotemplate and use default kubernetes values.

The default kubernetes values are : 
ipFamilyPolicy: "SingleStack"
ipFamilies: "" # no value, it depends on cluster configuration and ipFamilyPolicy

https://kubernetes.io/docs/reference/kubernetes-api/service-resources/service-v1/

**Comment by rdettai** [comment]
Sorry I didn't have the time to test this PR yet, I'll do it early next week. Sorry for the delay.

**Comment by rdettai** [comment]
I tested that it works without specifying the new parameters. I don't have a cluster with IPv6 configured so I can't test dual stack, but given that the current functionalities are not impact, I'm merging this anyway.

**Review by rdettai — APPROVED** [review]

**Pull request merged**